src/rivian/ble.py

```diff
@@ -0,0 +1,68 @@
+import platform
+import asyncio
+import secrets
+import logging
+
+from bleak import BleakClient, BLEDevice
+from rivian import utils
+
+_LOGGER = logging.getLogger(__name__)
+
+PHONE_ID_VEHICLE_ID_UUID = "AA49565A-4D4F-424B-4559-5F5752495445"
+PNONCE_VNONCE_UUID = "E020A15D-E730-4B2C-908B-51DAF9D41E19"
+BLE_ACTIVE_ENTRY_UUID = "5249565F-4D4F-424B-4559-5F5752495445"
+
+async def pair_phone(
+    device: BLEDevice, vehicle_id, phone_id, vehicle_key, private_key
+) -> bool:
+    success = True
+
+    # Create an asyncio.Event object to signal the arrival of a new notification.
+    notification_event = asyncio.Event()
+    notification_data = None
+
+    # Callback for notifications from vehicle characteristics 
+    def notification_handler(_, data: bytearray):
+        nonlocal notification_data
+        notification_data = data
+        notification_event.set()
+
+    try:
+        async with BleakClient(device, timeout=10) as client:
+            _LOGGER.debug(f"Connecting to {BLEDevice.name} [{BLEDevice.address}]")
+            await client.start_notify(PHONE_ID_VEHICLE_ID_UUID, notification_handler)
+            await client.start_notify(PNONCE_VNONCE_UUID, notification_handler)
+            # wait to enable notifications for BLE_ACTIVE_ENTRY_UUID
+
+            # write the phone ID (16-bytes) response will be vehicle ID
+            await client.write_gatt_char(PHONE_ID_VEHICLE_ID_UUID, bytes.fromhex(phone_id.replace("-", "")))
+            await notification_event.wait()
```

Comment:
We should have a timeout here to prevent an indefinite wait:
`await asyncio.wait_for(notification_event.wait(), 10)`

Reply:
10 seconds is probably too long, since we should get this result back pretty quickly if it is correct. So maybe set this to just a few seconds instead